src/fft.rg

```diff
@@ -144,133 +152,435 @@ function fft.generate_fft_interface(itype, dtype)
+    format.println("In iface.make_plan...")
+
+    --Obtain pointer to plan using get_plan
+    format.println("Calling get_plan...")
     var p = iface.get_plan(plan, false)
 
-    var address_space = c.legion_processor_address_space(
-      get_executing_processor(__runtime()))
+    --Get_executing process: takes a c.legion_runtime_t and returns c.legion_runtime_get_executing_processor(runtime, ctx)
+    var address_space = c.legion_processor_address_space(get_executing_processor(__runtime())) --legion_processor_address_space: takes a legion_processor_t proc and returns a legion_address_space_t
+
+    --Check input/output bounds. Bounds code example: var is = ispace(int1d, 12, -1). is.bounds returns rect1d { lo = int1d(-1), hi = int1d(10) }
+    regentlib.assert(input.ispace.bounds == output.ispace.bounds, "input and output regions must be identical in size")
+
+    --Get pointers to input and output regions
+    --get_base returns a base pointer to a region with ispace of dimension dim and fspace dtype. [local rect_t, get_base = make_get_base(dim, dtype)] [local terra get_base(rect : rect_t, physical : c.legion_physical_region_t, field : c.legion_field_id_t)]
+
+    var input_base = get_base_in(rect_t_in(input.ispace.bounds), __physical(input)[0], __fields(input)[0])
+    var output_base = get_base_out(rect_t_out(output.ispace.bounds), __physical(output)[0], __fields(output)[0])
+
+    -- Info on argumentes from https://legion.stanford.edu/doxygen/class_legion_1_1_physical_region.html
+    --__physical(r.{f, g, ...}) returns an array of physical regions (legion_physical_region_t) for r, one per field, for fields f, g, etc. in the order that the fields are listed in the call.
+    --__fields(r.{f, g, ...}) returns an array of the field IDs (legion_field_id_t) of r, one per field, for fields f, g, etc. in the order that the fields are listed in the call.
+
+    var lo = input.ispace.bounds.lo:to_point()
+    var hi = input.ispace.bounds.hi:to_point()
+
+    --dtype size is 8 for complex32 and 16 for complex64
+    format.println("Size of dtype is {}", dtype_size)
+
+
+     --Call fftw_c.fftw_plan_dft: fftw_plan_dft_1d(int n, fftw_complex *in, fftw_complex *out,int sign, unsigned flags). n is the size of transform, in and out are pointers to the input and output arrays. Sign is the sign of the exponent in the transform, can either be FFTW_FORWARD (1) or FFTW_BACKWARD (-1). Flags: FFTW_ESTIMATE, on the contrary, does not run any computation
+    format.println("Calling fftw_plan_dft to store fftw_plan in p.p...")
+
+    --R2C: Float to Complex32
+    if dtype_size == 8 and real_flag then
+      format.println("R2C: Float to Complex32: Not Supported")
+      --AK Note: I don't actually understand this that well, how the size of each dimension is stored in the 'n' array
+      var n : int[dim]
+      ;[data.range(dim):map(function(i) return rquote n[i] = hi.x[i] - lo.x[i] + 1 end end)] --n is a array of size dim storing the size of each dimension
```

Comment:
It seems like there's a bunch of redundant code like this. In general, when we don't need to duplicate code N times, it's better to avoid that because it makes everything else harder to understand.

Reply:
Pulled this out of the for-loops